data/proj.ini

```diff
@@ -15,6 +15,15 @@ cache_size_MB = 300
 
 cache_ttl_sec = 86400
 
+; Can be set to on so that by default the lack of a known resource files needed
+; for the best transformation PROJ would normally use causes an error. This
+; default value itself is overriden by the PROJ_ONLY_BEST_DEFAULT environment
+; variable if set, and then by the ONLY_BEST setting that can be
+; passed to the proj_create_crs_to_crs() method, or with the --only-best
+; option of the cs2cs program.
+; (added in PROJ 9.2)
+only_best_default = off
```

Comment:
Is there a way to have this unset? The reason is for warnings. If this is set to ON/OFF, the warnings should be disabled similar to the `ONLY_BEST` option passed into `proj_create_crs_to_crs_from_pj`, correct?

Reply:
should be fixed with latest commit (gizz, all those accumulation of settings gets crazy complicated)

Reply:
> (gizz, all those accumulation of settings gets crazy complicated)

Perhaps a sign that some refactoring is needed in the area? There must be a smart way to do it without plastering the whole code-base with tripwires

Reply:
It would be good if all problems had simple solutions :-)
To be honest, we're reaching a point where I'm losing interest spending more time working in this pull request.

Reply:
> To be honest, we're reaching a point where I'm losing interest spending more time working in this pull request.

Hey now, I was not suggesting that you include a major refactoring in this PR. If this is a pain point during development it is worth considering a solution to the problem. It should obviously be done in a separate issue but I don't want to open the discussion unless there's an actual problem. Hence my question.

Reply:
The only axis of simplification I can imagine would probably be to merge the 2 warn and error boolean variables in a single one with 3 states: no warning (only_best=off), warning (no explicit setting), error (only_best=on).
There isn't nothing that is particularly difficult in the code. There are places in PROJ that are > 10x times harder to make sense of ( ``CoordinateOperationFactory::Private::createOperationsBoundToGeog()`` probably holds the record... Or perhaps ``CoordinateOperationFactory::Private::createOperationsCompoundToGeog()``:-) )

Reply:
Alright, I'm glad we've de-escalated the situation from "crazy complicated" to "nothing .. particularly difficult" :-) My spidey-sense was tingling at the crazy complicated part - that sort of wording usually is an early warning that refactoring is in order. We don't always notice when knee-deep in the code so I just wanted to check whether this was an actual problem or not